### ISSUE TITLE
LM finetuning

### ISSUE THREAD

**Issue by bakszero** [facebookresearch/XLM#124]
Say, I want to finetune an XLM pretrained English model on a custom dataset for LM finetuning. What would be the steps to do that? Thanks!

**Comment by aconneau** [comment]
Hi @bakszero, the question is quite generic. What is your task? Current supported fine-tuning includes one-sentence (e.g. SST-2) or two-sentence (e.g. NLI) classification tasks, XNLI and MT. If your task is a classification task, I would follow the same procedure as for GLUE fine-tuning (see README) but modify the file here such that it can load your own task: https://github.com/facebookresearch/XLM/blob/master/src/evaluation/glue.py

**Comment by bakszero** [comment]
Thanks for your quick response, @aconneau. You're correct, I should have been more clear. 
What I'm attempting to do is use a fine-tuned XLM model as an encoder/decoder for a sequence to sequence task. Specifically, the fine-tuned XLM must be on some other domain than the dataset the initial XLM model was pre-trained on. This is similar to the ULMFiT paper, in that the model is fine-tuned on text from the same domain as the target task and using the "same pretraining objective".
For BERT, it is quite the same as https://github.com/huggingface/pytorch-pretrained-BERT/tree/master/examples/lm_finetuning and for GPT-2 @ this fork: https://github.com/nshepperd/gpt-2/   -- I'm looking for something like this. 

So my end goal is to use the finetuned(on the same pretraining objective, but with custom text) encoders and decoders as feeders for again finetuning on a sequence to sequence task. Do you think getting the pretrained XLM model finetuned on the same pretraining objective is possible right now? Thanks!

**Comment by aconneau** [comment]
Would reloading the XLM pretrained model via the ["--reload_model"](https://github.com/facebookresearch/XLM/blob/master/train.py#L180-L181) parameter, and using that as a starting point for an XLM training on new data do the trick? If so, this should be simple, and you should simply prepare your new data as specified in part I of the README, and then run a "python train.py ...." with the "--reload model" pointing to the pretrained model. Please let me know if I misunderstood.

**Comment by bakszero** [comment]
> Would reloading the XLM pretrained model via the ["--reload_model"](https://github.com/facebookresearch/XLM/blob/master/train.py#L180-L181) parameter, and using that as a starting point for an XLM training on new data do the trick? If so, this should be simple, and you should simply prepare your new data as specified in part I of the README, and then run a "python train.py ...." with the "--reload model" pointing to the pretrained model. Please let me know if I misunderstood.

Indeed , this would definitely help, thanks! 
One doubt though: My BPE codes and vocab would differ from the original corpus the model was trained on. Would this create a problem?

Also, the 2048 embedding-dim model seems too resource-heavy. Is it possible that you have a 512 or lower dimension one stored somewhere? Would be of great help, thanks!
